### PR TITLE
machine: make machinectl list[-images] --full work as expected

### PR DESCRIPTION
**Pull request by dongsupark**
Since systemd v239, both commands `machinectl list --full` and `machinectl list-images --full` printed out a shorted name for each machine name with an ellipsis character at the end. For example:
    
Before v239:
```
one-machine-with-a-long-name container systemd-nspawn - - 10.22.0.2...
```

Since v239:
```
one-machine-with-a-long-na… container systemd-nspawn - - 10.22.0.2…
```

Most users would have expected full machine names without ellipsis, when a `--full` option is given.

To make those commands print out full machine names, we need to make table-related functions take an option for full width. When the `--full` option is given, we calculate the greatest possible width for each column, to set the value to the display map of the table. That way we can make the table print out in a wide format.

See also https://github.com/kinvolk/kube-spawn/pull/317

### CODE REVIEW

File: src/login/loginctl.c [hunk 1]
```diff
@@ -282,14 +282,15 @@ static int list_seats(int argc, char *argv[], void *userdata) {
...
 
                 r = sd_bus_message_read(reply, "(so)", &seat, NULL);
                 if (r < 0)
                         return bus_log_parse_error(r);
                 if (r == 0)
                         break;
 
-                r = table_add_cell(table, NULL, TABLE_STRING, seat, arg_full);
+                r = table_add_cell(table, &cell, TABLE_STRING, seat, arg_full);
                 if (r < 0)
```
Comment (keszybz) [inline src/login/loginctl.c:298]: Why is this change needed?

File: src/basic/format-table.c [hunk 2]
```diff
@@ -586,7 +601,7 @@ int table_add_many_internal(Table *t, bool is_full, TableDataType first_type, ..
                         assert_not_reached("Uh? Unexpected data type.");
                 }
 
-                r = table_add_cell(t, NULL, type, data, is_full);
+                r = table_add_cell(t, &cell, type, data, is_full);
```
Comment (keszybz) [inline src/basic/format-table.c]: `cell` isn't used afterwards.

File: src/login/loginctl.c (hunk 1 above)
Comment (dongsupark) [reply src/login/loginctl.c:298]: Right, I removed the unnecessary cell variables.

**Comment by dongsupark** [comment]
@keszybz Thanks for testing it.
Indeed this PR made other bugs, so I've just fixed them.
To catch such bugs, I added another test `test_narrow_width()` that does the test you described.
Also did some refactoring, added missing error checks, removed unnecessary cell variables, etc.

**Comment by keszybz** [comment]
Hmm, when I merged this into current master, I see:
```console
$ build/machinectl --no-pager list
MACHINE                    CLASS     SERVICE        OS                         VERSION ADDRESSES      
debian-unstable            container systemd-nspawn debian                     -       -              
rawhide-very-long-name-la… container systemd-nspawn fedoraaaaaaaaaaaaaaaaaaaa… 30      169.254.89.142…

2 machines listed.
$ build/machinectl --no-pager list --full
MACHINE                                               CLASS      SERVICE         OS                                                          VERSION ADDRESSES         
debian-unstable                                       container  systemd-nspawn  debian                                                      -       -                 
rawhide-very-long-name-lasdfasdfasdfasdfasdfasdfasdf  container  systemd-nspawn  fedoraaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa  30      169.254.89.142…   

2 machines listed.
```
It looks like there's an off-by-one error in the calculation of widths (an extra empty column is inserted).

**Comment by poettering** [comment]
(i figure we need to teach format-table properly multiline-cell support. would love to add that fully, but i'd really see #10701 merged first, which touches a lot of the table code already)

**Comment by dongsupark** [comment]
Rebased.

@keszybz Yeah, I'm aware of that issue. You can also see the extra columns in [the expected output in a new test](https://github.com/systemd/systemd/pull/10110/commits/10490859cb2cc8507de9d5a481c643572ae88352#diff-6518239bd110ce841782ed1e7f341fdcR76). That column width is inserted [here](https://github.com/systemd/systemd/pull/10110/commits/10490859cb2cc8507de9d5a481c643572ae88352#diff-f1c6c40e3b339e453e89f1ef0dd1c86cR381).

The thing is, if the new extra column is not inserted, then the name column is ellipsized again, even if there's enough space. Obviously that was not exactly what I wanted to achieve. I suppose, that issue has something to do with the calculation of weighted column widths in [`table_print()`](https://github.com/systemd/systemd/blob/09e5c07fbd23d098edf2e0b51fbf402f07474dae/src/shared/format-table.c#L1033). I have not been able to exactly figure out why that happens.
Anyway I thought, if a user passed an option `--full`, it's more reasonable to print out everything (including the additional columns), than printing out an ellipsis in the actual columns.

If you still think that the extra column needs to be removed, please let me know. Then I can try again to debug it. I'm fine even if this PR would not be merged before the next release.

**Comment by keszybz** [comment]
I think we need to figure out where this extra column is coming from and fix that. It's a pretty-clear off-by-one somewhere. This also needs a rebase now ;(

**Comment by poettering** [comment]
This should be fixed since a65e34ccb083a03aa8ecbcbf12fc59349802d014 and related commits. Closing.
